test/mode/modeInsert.test.ts

```diff
@@ -625,4 +625,38 @@ suite('Mode Insert', () => {
       endMode: Mode.Insert,
     });
   });
+
+  suite('<C-g>', () => {
```

Comment:
Going along with behavior change I mentioned above, I'd want to see tests with insertions, deletions, and arrow key movements before the `<C-g>` movement